libtcmu.c

```diff
@@ -590,6 +607,10 @@ static void remove_device(struct tcmulib_context *ctx,
 	if (ret < 0)
 		tcmu_err("could not cleanup caw lock %s: %d\n", dev_name, errno);
 
+	ret = pthread_mutex_destroy(&dev->format_lock);
+	if (ret < 0)
+		tcmu_err("could not cleanup format lock %s: %d\n", dev_name, errno);
```

Comment:
I feel like  must be looking at something wrong. This is what I have for my pthread_mutex_destroy man page:

RETURN VALUE
       If  successful,  the  pthread_mutex_destroy()  and pthread_mutex_init()
       functions shall return  zero;  otherwise,  an  error  number  shall  be
       returned to indicate the error.

Normally, when errno is set, I thought it says "and errno is set appropriately", but I do not see that. However, you and Venky (he added the chunk above your code), seem to see something different. Does it mention -1 and errno in your man page.